components/net/cookie_storage.rs

```diff
@@ -229,14 +229,19 @@ fn reg_host<'a>(url: &'a str) -> String {
 
 fn is_cookie_expired(cookie: &Cookie) -> bool {
     match cookie.expiry_time {
-        Some(ref t) => t.to_timespec() <= time::get_time(),
+        Some(ref t) => {
+            t.duration_since(UNIX_EPOCH).unwrap_or_default() <=
+                SystemTime::now()
+                    .duration_since(UNIX_EPOCH)
+                    .unwrap_or_default()
+        },
         None => false,
     }
 }
 
 fn evict_one_cookie(is_secure_cookie: bool, cookies: &mut Vec<Cookie>) -> bool {
     // Remove non-secure cookie with oldest access time
-    let oldest_accessed: Option<(usize, Tm)> = get_oldest_accessed(false, cookies);
+    let oldest_accessed: Option<(usize, SystemTime)> = get_oldest_accessed(false, cookies);
```

Comment:
I think you can omit the type here and below. The compiler should be able to figure this out, right?

```suggestion
    let oldest_accessed = get_oldest_accessed(false, cookies);
```

Reply:
Ah yes indeed